Code/GraphMol/MolStandardize/Pipeline.h

```diff
@@ -0,0 +1,231 @@
+                      STEREO_VALIDATION_ERROR),
+  PREPARE_FOR_STANDARDIZATION_ERROR = (1 << 7),
+  METAL_STANDARDIZATION_ERROR = (1 << 8),
+  NORMALIZER_STANDARDIZATION_ERROR = (1 << 9),
+  FRAGMENT_STANDARDIZATION_ERROR = (1 << 10),
+  CHARGE_STANDARDIZATION_ERROR = (1 << 11),
+  STANDARDIZATION_ERROR =
+      (METAL_STANDARDIZATION_ERROR | NORMALIZER_STANDARDIZATION_ERROR |
+       FRAGMENT_STANDARDIZATION_ERROR | CHARGE_STANDARDIZATION_ERROR),
+  OUTPUT_ERROR = (1 << 12),
+  PIPELINE_ERROR = (INPUT_ERROR | PREPARE_FOR_VALIDATION_ERROR |
+                    VALIDATION_ERROR | PREPARE_FOR_STANDARDIZATION_ERROR |
+                    STANDARDIZATION_ERROR | OUTPUT_ERROR),
+  METALS_DISCONNECTED = (1 << 23),
+  NORMALIZATION_APPLIED = (1 << 24),
+  FRAGMENTS_REMOVED = (1 << 25),
+  PROTONATION_CHANGED = (1 << 26),
+  STRUCTURE_MODIFICATION = (METALS_DISCONNECTED | NORMALIZATION_APPLIED |
+                            FRAGMENTS_REMOVED | PROTONATION_CHANGED)
+};
+
+enum class RDKIT_MOLSTANDARDIZE_EXPORT PipelineStage : std::uint32_t {
+  NOT_STARTED = 0,
+  PARSING_INPUT,
+  PREPARE_FOR_VALIDATION,
+  VALIDATION,
+  PREPARE_FOR_STANDARDIZATION,
+  STANDARDIZATION,
+  REAPPLY_WEDGING,
+  CLEANUP_2D,
+  MAKE_PARENT,
+  SERIALIZING_OUTPUT,
+  COMPLETED
+};
+
+struct RDKIT_MOLSTANDARDIZE_EXPORT PipelineLogEntry {
+  PipelineStatus status;
+  std::string detail;
+};
+
+using PipelineLog = std::vector<PipelineLogEntry>;
+
+struct RDKIT_MOLSTANDARDIZE_EXPORT PipelineResult {
+  PipelineStatus status;
+  std::uint32_t stage;
+  PipelineLog log;
+  std::string inputMolData;
+  std::string outputMolData;
+  std::string parentMolData;
+
+  void append(PipelineStatus newStatus, const std::string& info);
+};
+
+using RWMOL_SPTR_PAIR = std::pair<RWMOL_SPTR, RWMOL_SPTR>;
+
+namespace Operations {
+RWMOL_SPTR prepareForValidation(RWMOL_SPTR mol, PipelineResult& result,
+                                const PipelineOptions& options);
+RWMOL_SPTR validate(RWMOL_SPTR mol, PipelineResult& result,
+                    const PipelineOptions& options);
+RWMOL_SPTR prepareForStandardization(RWMOL_SPTR mol, PipelineResult& result,
+                                     const PipelineOptions& options);
+RWMOL_SPTR standardize(RWMOL_SPTR mol, PipelineResult& result,
+                       const PipelineOptions& options);
+RWMOL_SPTR reapplyWedging(RWMOL_SPTR mol, PipelineResult& result,
+                          const PipelineOptions& options);
+RWMOL_SPTR cleanup2D(RWMOL_SPTR mol, PipelineResult& result,
+                     const PipelineOptions& options);
+RWMOL_SPTR_PAIR makeParent(RWMOL_SPTR mol, PipelineResult& result,
+                           const PipelineOptions& options);
+
+RWMOL_SPTR parse(const std::string& molblock, PipelineResult& result,
+                 const PipelineOptions& options);
+void serialize(RWMOL_SPTR_PAIR output, PipelineResult& result,
+               const PipelineOptions& options);
```

Comment:
This change should get the windows DLL builds working.
Sorry I forgot to do this in my PR.

```suggestion
RDKIT_MOLSTANDARDIZE_EXPORT RWMOL_SPTR prepareForValidation(
    RWMOL_SPTR mol, PipelineResult &result, const PipelineOptions &options);
RDKIT_MOLSTANDARDIZE_EXPORT RWMOL_SPTR validate(RWMOL_SPTR mol,
                                                PipelineResult &result,
                                                const PipelineOptions &options);
RDKIT_MOLSTANDARDIZE_EXPORT RWMOL_SPTR prepareForStandardization(
    RWMOL_SPTR mol, PipelineResult &result, const PipelineOptions &options);
RDKIT_MOLSTANDARDIZE_EXPORT RWMOL_SPTR standardize(
    RWMOL_SPTR mol, PipelineResult &result, const PipelineOptions &options);
RDKIT_MOLSTANDARDIZE_EXPORT RWMOL_SPTR reapplyWedging(
    RWMOL_SPTR mol, PipelineResult &result, const PipelineOptions &options);
RDKIT_MOLSTANDARDIZE_EXPORT RWMOL_SPTR cleanup2D(
    RWMOL_SPTR mol, PipelineResult &result, const PipelineOptions &options);
RDKIT_MOLSTANDARDIZE_EXPORT RWMOL_SPTR_PAIR makeParent(
    RWMOL_SPTR mol, PipelineResult &result, const PipelineOptions &options);

RDKIT_MOLSTANDARDIZE_EXPORT RWMOL_SPTR parse(const std::string &molblock,
                                             PipelineResult &result,
                                             const PipelineOptions &options);
RDKIT_MOLSTANDARDIZE_EXPORT void serialize(RWMOL_SPTR_PAIR output,
                                           PipelineResult &result,
                                           const PipelineOptions &options);
```

Reply:
Thank you, I also didn't think about this until I saw the CI errors.